python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -97,7 +101,25 @@ def op_config(self) -> Any:
         """The parsed config specific to this op."""
 
 
-class OpExecutionContext(AbstractComputeExecutionContext):
+class OpExecutionContextMetaClass(AbstractComputeMetaclass):
+    def __instancecheck__(cls, instance) -> bool:
+        # This makes isinstance(context, OpExecutionContext) throw a deprecation warning when
+        # context is an AssetExecutionContext. This metaclass can be deleted once AssetExecutionContext
+        # has been split into it's own class in 1.7.0
+        if isinstance(instance, AssetExecutionContext):
```

Comment:
what's the right way to have this deprecation warning on `isintance(obj, OpExecutionContext`? BK is showing recursion errors related to doing `isinstance(instance, AssetExecutionContext)` since that in turn has this overriden instancecheck

```


[2023-09-18T18:26:47Z] dagster/_core/definitions/op_definition.py:449: in __call__
--
  | [2023-09-18T18:26:47Z]     return direct_invocation_result(self, *args, **kwargs)
  | [2023-09-18T18:26:47Z] dagster/_core/definitions/op_invocation.py:172: in direct_invocation_result
  | [2023-09-18T18:26:47Z]     elif len(args) > 0 and isinstance(args[0], UnboundOpExecutionContext):
  | [2023-09-18T18:26:47Z] dagster/_core/execution/context/compute.py:109: in __instancecheck__
  | [2023-09-18T18:26:47Z]     if isinstance(instance, AssetExecutionContext):
  | [2023-09-18T18:26:47Z] dagster/_core/execution/context/compute.py:109: in __instancecheck__
  | [2023-09-18T18:26:47Z]     if isinstance(instance, AssetExecutionContext):
  | [2023-09-18T18:26:47Z] dagster/_core/execution/context/compute.py:109: in __instancecheck__
  | [2023-09-18T18:26:47Z]     if isinstance(instance, AssetExecutionContext):
  | [2023-09-18T18:26:47Z] E   RecursionError: maximum recursion depth exceeded while calling a Python object
  | [2023-09-18T18:26:47Z] !!! Recursion detected (same locals & position)


```

Reply:
I believe the issue here is that you have to do an exact type match on `AssetExecutionContext` by using `type` or something, rather than `isinstance`. Looks the metaclass instance invokes itself when `isinstance` is used..

Reply:
ah cool that makes sense